### PR TITLE
Refactor MemoryTest out of Stats namespace

### PR DESCRIPTION
**Pull request by ravenblackx**
Commit Message: Refactor MemoryTest out of Stats namespace
Additional Description: MemoryTest makes more sense in Memory namespace. It's already used in various not-stats contexts. Splitting the header also reduces the total amount of header pulled in by some tests.
Risk Level: None, it's test-only *and* a practical no-op.
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: test/common/stats/stat_test_utility.h [hunk 1]
```diff
@@ -57,54 +56,6 @@ class TestSymbolTable {
...
                        std::function<void(absl::string_view)> fn);
 
-// Tracks memory consumption over a span of time. Test classes instantiate a
-// MemoryTest object to start measuring heap memory, and call consumedBytes() to
-// determine how many bytes have been consumed since the class was instantiated.
-//
-// That value should then be passed to EXPECT_MEMORY_EQ and EXPECT_MEMORY_LE,
-// defined below, as the interpretation of this value can differ based on
-// platform and compilation mode.
-class MemoryTest {
```
Comment (jmarantz) [inline test/common/stats/stat_test_utility.h:67]: can you do me a favor and leave  a `using` nickname for MemoryTest here, pointing to its new location?

We have a bunch of references to the current name in non-OSS code that we can fix to point to your new one, and then after htat rmeove the nickname. You can leave an issue assigned to me to clean this up.

Comment (ravenblackx) [reply test/common/stats/stat_test_utility.h:67]: Done.

**Review by jmarantz — APPROVED** [review]

**Comment by jmarantz** [comment]
format failure 

/wait

**Review by ggreenway — APPROVED** [review]

**Pull request merged**